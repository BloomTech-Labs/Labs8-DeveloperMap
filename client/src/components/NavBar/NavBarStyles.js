import styled from 'styled-components';

export const Nav = styled.div`
  width: 98%;
  height: 50px;
  background-color: rgba(232, 232, 232, .85);
  z-index: 1;
  position: absolute;
  border-radius: 25px;
  display: flex;
  top: 6px;
  right: 1%;
  justify-content: flex-end;
  align-items: center;
  box-shadow: inset 10px 10px 10px rgba(255,255,255,.5);
  .heart {
    margin-right: 2%;
    img {
      width: 25px;
      height: 25px;
    }
  }
  .avatar {
    
    img {
      border-radius: 50%;
      width: 27px;
      height: 27px;
    }
  }
  
`;


export const Icons = styled.div`
  display: flex;
  margin: 0 2%;
  justify-content: space-between;
  width: 100px;
  .signout{
    img {
      border-radius: 50%;
      width: 27px;
      height: 27px;
    } 
  }
`;

export const InButton = styled.button`
  width: 75px;
  height: 35px;
  background-color: rgb(184,15,10);
  color: white;
  border-radius: 20px;
  border: none;
  margin-right: 10px;
  box-shadow: 0 4px 2px -2px gray;
  &:hover {
    background-color: rgba(184,15,10,.8);
  } 
  &:focus {
    outline: 0;
  }
`;