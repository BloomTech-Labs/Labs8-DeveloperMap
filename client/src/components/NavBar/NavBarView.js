import React from 'react';
import Heart from '../../images/favheart.png';
import Avatar from '../../images/gear.png';
import Logout from '../../images/logout1.png';
import { Nav, Icons, InButton } from './NavBarStyles';

class NavBar extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
  }
  handleSignIn = () => {
    this.props.history.push('/signin');
  };

  favorite = () => {
    const { uid } = this.props.user;
    this.props.history.push(`/favorites/${this.props.user.uid}`);
  };

  profilePop = () => {
    this.props.history.push(`/settings`);
  };

  render() {
    return (
      <Nav>
        {this.props.user ? (
          <Icons>
            <div className="heart" onClick={this.favorite}>
              <img alt="Favorites Icon" src={Heart} title="Favorites" />
            </div>
            <div className="avatar">
              <img
                alt="Avatar Icon"
                src={Avatar}
                onClick={this.profilePop}
                title="User Settings"
              />
            </div>
            <div className="signout">
              <img
                alt="Logout"
                src={Logout}
                onClick={this.props.signOut}
                title="Logout"
              />
            </div>
          </Icons>
        ) : (
          <div>
            <InButton onClick={this.handleSignIn}>Sign In ></InButton>
          </div>
        )}
      </Nav>
    );
  }
}

export default NavBar;
