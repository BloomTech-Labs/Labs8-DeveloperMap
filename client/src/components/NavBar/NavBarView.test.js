import React from 'react';
import renderer from 'react-test-renderer';
import NavBar from './NavBarView';


describe(`Nav Bar Component`, () => {
    it('renders correctly', () => {
        const tree = renderer.create(
          <NavBar />
        ).toJSON();
        expect(tree).toMatchSnapshot();
      });

    it('render user prop correctly with null value', () => {  
      const props = {
              user: null
          },
          NavComponent = mount(<NavBar {...props} />);
      expect((NavComponent).prop('user')).toEqual(null);
    });  

    it('render user with empty value', () => {  
        const props = {
            user: ''
        },
        NavComponent = mount(<NavBar {...props} />);
    expect((NavComponent).prop('user')).toBeFalsy();
    });

})