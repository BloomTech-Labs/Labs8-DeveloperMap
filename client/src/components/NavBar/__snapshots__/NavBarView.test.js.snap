// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav Bar Component renders correctly 1`] = `
<div
  className="nav sc-bdVaJa gzaecn"
>
  <div>
    <button
      className="sign-in sc-htpNat dJuGll"
      onClick={[Function]}
    >
      Sign In &gt;
    </button>
  </div>
</div>
`;
